### DISCUSSION TITLE
running the project

### DISCUSSION THREAD

**Discussion by cweave** [origranot/reduced.to#194] [Q&A] [answered]
I'm still a little confused on the frontend application. The application that initially loads at `http://127.0.0.1:5173/` is not the actual frontend app and doesn't have any of the baked in HMR or anything. 

```bash
cd frontend
npm run start
```

Also I'm having issues running prod, `npm run build:all` then `npm run start:prod`
![image](https://user-images.githubusercontent.com/16121690/195496252-c9fd5594-8de8-4eb7-8d94-d1a8b8d7e156.png)

**Comment by GuillemPM** [comment, marked as answer]
Please run npm install && npm run install:all from the root as it is in the docs and try again

Follow our README
https://github.com/origranot/url-shortener#-installation

**Reply by cweave** [reply]
yep, did that.

**Reply by GuillemPM** [reply]
Be sure to sync all the changes from our repo. To do that from github: go to your fork, master branch and click on the sync button 
![image](https://user-images.githubusercontent.com/26607131/195642809-baf23ade-28bd-4ee3-b67e-1591f303a1d8.png)

**Reply by cweave** [reply]
![image](https://user-images.githubusercontent.com/16121690/195728784-ee0e4b2d-28e1-4f17-ae7a-df1ccdf38dc0.png)

Did a fresh clone off of master on a different computer. This was `npm install && npm run install:all` following along to the README.

Is it not super friendly with windows?

**Reply by GuillemPM** [reply]
Looks like a problem with Vite, which version of node and npm are you using?

I develop on windows so that shouldn't be a problem

**Reply by cweave** [reply]
node v16.1.1, npm v8.15.1. Does node need to be v18? I was having issues with Vite v3 and node versions at work.

**Reply by detronetdip** [reply]
I think you have to upgrade your nodejs otherwise you can use docker or docker-compose too for running the project locally. you can find all the instruction in the readme file.

**Reply by GuillemPM** [reply]
I'm currently working on node 16.14.0 and npm 8.3.1, not sure if your node version should be at least 16.14+
I'll try to check

**Reply by GuillemPM** [reply]
It should be compatible with any version >= 16.0.0
If you run the command npm run start in frontend folder is giving the same error? or only during building for production?

**Reply by cweave** [reply]
per the documentation,
![image](https://user-images.githubusercontent.com/16121690/196308965-cdb042e9-2c6d-4edf-94d9-97bb7a6145de.png)

but `npm run start` gets it up.

**Reply by GuillemPM** [reply]
Don't try to run npm run start:dev in the frontend folder, it is a backend command.
Looks like you having problems installing all the dependencies, (reflect metadata is missing). Please be sure to install all the dependencies needed, if your node can't handle it, do it manually.
